play-json/jvm/src/main/scala/play/api/libs/json/jackson/JacksonJson.scala

```diff
@@ -282,9 +285,17 @@ private[json] object JacksonJson {
 }
 
 private[json] case class JacksonJson(jsonConfig: JsonConfig) {
-  private val mapper = (new ObjectMapper).registerModule(new PlayJsonMapperModule(jsonConfig))
-
-  private val jsonFactory = new JsonFactory(mapper)
+  private val streamReadConstraints = StreamReadConstraints
+    .builder()
+    .maxNumberLength(Integer.MAX_VALUE)
+    .build()
+  private val jsonFactory = JsonFactory
+    .builder()
+    .asInstanceOf[JsonFactoryBuilder]
+    .streamReadConstraints(streamReadConstraints)
+    .recyclerPool(JsonRecyclerPools.threadLocalPool())
```

Comment:
Jackson 2.17.0 used a very inefficient recycler pool that behaved even worse under heavy load - called 'Lock Free'. Jackson 2.17.1 and 2.17.2 (just released) goes back to ThreadLocalPool as the default. Still maybe a good idea to hardcode its use though.